test/issuehunter.js

```diff
@@ -71,16 +85,48 @@ const addressBalance = Promise.denodeify(function (address, callback) {
 
 const gasPrice = Promise.denodeify(web3.eth.getGasPrice)
 
-const newCampaignId = (function () {
-  var counter = 200
+const nextCampaignId = (function () {
+  var counter = 0
   return function () {
     counter += 1
     return `new-campaign-${counter}`
   }
 })()
 
+const sample = function (array) {
+  return function () {
+    return array[Math.floor(Math.random() * array.length)]
+  }
+}
+
+const sampleExcluding = function (array) {
+  return function (exclusions) {
+    const filtered = array.filter(function (item) {
+      return exclusions.indexOf(item) < 0
+    })
+    return sample(filtered)()
+  }
+}
+
+// TODO: find a better way to check for a user's account balance delta This is a
+// workaround and it won't work under all conditions. It partially works because
+// transactions are in wei, but gas are some orders of magnitude more expensive.
+// There's also an ugly hack to account for the fact that sometimes it could
+// happen that `initial + expected delta > modulo`, in those case the result
+// must be adjusted by `modulo`.
+const hackDelta = function (current, initial, expectedDelta) {
```

Comment:
I need more context here, please.

Reply:
When I assert that account balances amounts changed, I started using `mod(10000)` to factor out gas expenses. The `mod(10000)` factors out gas expenses because transactions values used in tests are in the order of tens of wei, whereas gas expenses are in the order of 10,000,000,000 wei, see https://github.com/ethereumjs/testrpc#command-line.

There are some cases though when an account balance initial amount `mod(10000)`, plus the expected delta, returns a value that is greater than 10000, screwing up the expected test result.

For instance if the account balance initial amount % 10000 is 9990 and the expected delta is 26, the final amount would be ...9990 + 26 = ...10016, so the difference between final amount and initial amount would be (...10016 % 10000) - (...9990 % 10000) = -9974, that isn't 26, the expected delta value.

In these cases, when initial amount + expected delta > 10000, an easy solution is to add 10000 to the result, for instance -9974 + 10000 = 26.